### DISCUSSION TITLE
Error reporting for hidden cells (OJS)

### DISCUSSION THREAD

**Discussion by declann** [quarto-dev/quarto-cli#8326] [Feature Requests]
### Description

Hi,

I'm developing a dashboard on Quarto `1.4.543` and the output quality is really nice! I like this feature a lot! However it's also drawn me to using `//| output: false` for the first time on OJS cells.

I had real hell getting to the bottom of some buggy behavior (at least twice) and was questioning what was going on.

It turns out I'm not seeing OJS error messages on account of my output being hidden, hiding the source of all my bugs! Error messages also don't appear in console.

This won't catch me so bad again, but should something be added to make this error reporting better? Is console logging OJS error messages reasonable and sufficient (I think so)? Is there a related issue for non-OJS? (I guess lower execution cell interaction, so errors would be more obvious?)

Thanks!
Declan

**Comment by cscheid** [comment]
> It turns out I'm not seeing OJS error messages on account of my output being hidden

This is unfortunate, but because of the way OJS works, when you have a cell with `output: false` we lack the ability to detect where the error happened, which we need in order to report it on the webpage. There are solutions we could try to implement, but none of them are great. With that said, I _am_ surprised that you're getting no console messages either. 

Can you point me to a concrete example of what was happening?

**Reply by declann** [reply]
Here is a simple example:

````
--- 
title: "Development Indicators by Continent"
author: "Gapminder Analytics Group"
format: dashboard
--- 

```{ojs}
2+2
```

## Row {height=60%}

```{ojs}
//| title: GDP and Life Expectancy 
3+3 
```

```{ojs}
//| output: false

document.getElementById('fdsafds').textContent = 'fewraf' // <----- THIS IS AN ERROR
```

## Row {height=40%}

```{ojs}
//| title: Population
2+2
```
````

There is an error, but nothing logged to console.

![image](https://github.com/quarto-dev/quarto-cli/assets/353214/9b4de17c-9142-417e-bc17-6061cf2b0f39)


If I show the cell:

![image](https://github.com/quarto-dev/quarto-cli/assets/353214/89e5f7d1-9ef1-46c2-9e1c-9909caf08af7)

**Reply by declann** [reply]
A console message would be absolutely perfect. In my actual case I am doing hacky stuff like updating DOM elements that are overlayed on CodeMirror, which I'm never coded before, so I had many bits of code to doubt. And I know that I'm also using a new/beta feature on Quarto, and I thought there could be issues with the runtime.

But actually, my code was buggy - which is a good outcome! But a console error would have set me straight much sooner :smile:

**Reply by cscheid** [reply]
Yes, I can repro this, and it's easy to do much better. It's too late for Quarto 1.4, but keep an eye for 1.5 development starting hopefully early next week.

**Reply by declann** [reply]
Excellent. Thanks Carlos!

**Comment by mcanouil** [comment]
Closing in favour of:
- https://github.com/quarto-dev/quarto-cli/issues/8327
